### DISCUSSION TITLE
Choose which core(s) effect/slots should be processed on.

### DISCUSSION THREAD

**Discussion by 38github** [rerdavies/pipedal#100] [Ideas]
This might be a big thing to ask for but would it be possible to choose which core the plugin/pedal slot is going to be processed on? Why I am asking is because on my four core SBC all cores except one is almost idle but Pipedal then use almost all the CPU of the remaining core.

Neural Amp Modeler is using about 50-60% CPU and the IR Cab plugin use about 30% (very short IR's). Placing these things on seperate cores would, in my belief, allow more high quality effects and better experience.

**Comment by rerdavies** [comment]
Multi-core processing is more useful in flow-blown DAWs where entire tracks and their effects chains can be run on separate threads, usually with very substantial latency. It's not so useful for Guitar effects pedals.

Because the effects are executed serially, you get no advantage from using multiple cores. The output of one effects is fed to the input of the next effect, which can't do anything until the first effect has generated output. If they ran on separate threads the second effect would just be waiting for the first effect to complete.

There is a corner case where two heavy-CPU effects are on parallel splits. (e.g. different amp models on left and right channels). But I'm not honestly sure that it's worth the effort and substantial risk of additional audio dropouts. The OS calls to do the necessary context switches aren't free.  And I don't imagine that's really what you want.

**Reply by 38github** [reply]
Thank you for the detailed explaination! I understand now why it is not feasable. Regards
